### PR TITLE
Tckfatproject

### PR DESCRIPTION
**Pull request by janamanoharan**

### CODE REVIEW

File: dev/com.ibm.ws.microprofile.openapi_fat_tck/bnd.bnd [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+# To define a global minimum java level for the FAT, use the following property.
+# If unspecified, the default value is ${javac.source}
+runtime.minimum.java.level: 1.8
+Require-Capability: osgi.ee;filter:="(&(osgi.ee=JavaSE)(version=1.8))"
+
+# Dependencies may be local bundles (e.g. com.ibm.websphere.javaee.servlet.3.1)
+#      or binaries from Artifactory (e.g. org.apache.derby:derbynet)
+# For all project names that match the pattern "*_fat*", dependencies for junit,
+# fattest.simplicity, and componenttest-1.0 will be automatically added to the buildpath
+-buildpath: \
```
Comment (janamanoharan) [inline dev/com.ibm.ws.microprofile.openapi_fat_tck/bnd.bnd]: commons-lang3 in Liberty (done)

File: dev/com.ibm.ws.microprofile.openapi_fat_tck/fat/src/org/eclipse/microprofile/openapi/fat/tck/ConfigTckPackageTest.java [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+import componenttest.topology.impl.LibertyServer;
+import componenttest.topology.utils.MvnUtils;
+import com.ibm.websphere.simplicity.PortType;
+/**
+ * This is a test class that runs a whole Maven TCK as one test FAT test.
+ * There is a detailed output on specific
+ */
+@SuppressWarnings("restriction")
+@RunWith(FATRunner.class)
+public class ConfigTckPackageTest {
```
Comment (janamanoharan) [inline dev/com.ibm.ws.microprofile.openapi_fat_tck/fat/src/org/eclipse/microprofile/openapi/fat/tck/ConfigTckPackageTest.java]: Change name to OpenAPITckTest (done)

File: dev/com.ibm.ws.microprofile.openapi_fat_tck/fat/src/org/eclipse/microprofile/openapi/fat/tck/ConfigTckPackageTest.java [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+
```
Comment (janamanoharan) [inline dev/com.ibm.ws.microprofile.openapi_fat_tck/fat/src/org/eclipse/microprofile/openapi/fat/tck/ConfigTckPackageTest.java]: remove empty lines (done)

File: dev/com.ibm.ws.microprofile.openapi_fat_tck/fat/src/org/eclipse/microprofile/openapi/fat/tck/ConfigTckPackageTest.java [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+    public static LibertyServer server;
+
+    @BeforeClass
+    public static void setUp() throws Exception {
+        server.startServer();
+    }
+
+    @AfterClass
+    public static void tearDown() throws Exception {
+        server.stopServer("CWMCG0007E", "CWMCG0014E", "CWMCG0015E", "CWMCG5003E", "CWWKZ0002E");
```
Comment (janamanoharan) [inline dev/com.ibm.ws.microprofile.openapi_fat_tck/fat/src/org/eclipse/microprofile/openapi/fat/tck/ConfigTckPackageTest.java]: remove the errors and test (done)

File: dev/com.ibm.ws.microprofile.openapi_fat_tck/fat/src/org/eclipse/microprofile/openapi/fat/tck/FATSuite.java [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+ *******************************************************************************/
+package org.eclipse.microprofile.openapi.fat.tck;
+
+import org.junit.runner.RunWith;
+import org.junit.runners.Suite;
+import org.junit.runners.Suite.SuiteClasses;
+
+@RunWith(Suite.class)
+@SuiteClasses({
+                ConfigTckPackageTest.class
```
Comment (janamanoharan) [inline dev/com.ibm.ws.microprofile.openapi_fat_tck/fat/src/org/eclipse/microprofile/openapi/fat/tck/FATSuite.java]: change name (done)

File: dev/com.ibm.ws.microprofile.openapi_fat_tck/fat/src/org/eclipse/microprofile/openapi/fat/tck/ConfigTckPackageTest.java [hunk 6]
```diff
@@ -0,0 +1,88 @@
...
+    }
+
+    @Test
+    @AllowedFFDC // The tested deployment exceptions cause FFDC so we have to allow for this.
+    public void testConfigTck() throws Exception {
```
Comment (navidsh) [inline dev/com.ibm.ws.microprofile.openapi_fat_tck/fat/src/org/eclipse/microprofile/openapi/fat/tck/ConfigTckPackageTest.java]: Rename this method to maybe `testOpenAPITck`

File: dev/com.ibm.ws.microprofile.openapi_fat_tck/publish/tckRunner/tck/README.md [hunk 7]
```diff
@@ -0,0 +1,39 @@
+# Instructions for Tck Runner
```
Comment (janamanoharan) [inline dev/com.ibm.ws.microprofile.openapi_fat_tck/publish/tckRunner/tck/README.md]: don't need this file (removed)

File: dev/com.ibm.ws.microprofile.openapi_fat_tck/publish/tckRunner/tck/pom.xml [hunk 8]
```diff
@@ -0,0 +1,84 @@
...
+    <groupId>com.ibm.ws.microprofile.openapi</groupId>
+    <artifactId>tck.runner.tck</artifactId>
+    <version>1.0-SNAPSHOT</version>
+    <packaging>jar</packaging>
+
+    <name>MicroProfile OpenAPI TCK Runner</name>
+    <url>https://github.ibm.com/</url>
+
+    <properties>
+        <microprofile.openapi.version>1.1-SNAPSHOT</microprofile.openapi.version>
```
Comment (janamanoharan) [inline dev/com.ibm.ws.microprofile.openapi_fat_tck/publish/tckRunner/tck/pom.xml]: use 1.0 as version number

File: dev/com.ibm.ws.microprofile.openapi_fat_tck/fat/src/org/eclipse/microprofile/openapi/fat/tck/ConfigTckPackageTest.java [hunk 9]
```diff
@@ -0,0 +1,88 @@
...
+        } catch (java.nio.file.NoSuchFileException nsfe) {
+            Assert.assertNull(
+                    "The TCK tests' results directory does not exist which suggests the TCK tests did not run - check build logs."
+                            + src.getAbsolutePath(), nsfe);
+        }
+        
+        // mvn returns 0 if all surefire tests pass and -1 otherwise - this Assert is enough to mark the build as having failed
+        // the TCK regression
+
+//        Assert.assertTrue(server.getInstallRoot() + "  com.ibm.ws.microprofile.openapi_fat_tck:org.eclipse.microprofile.openapi.fat.tck.ConfigTckPackageTest:testTck:TCK has returned non-zero return code of: " + rc
```
Comment (navidsh) [inline dev/com.ibm.ws.microprofile.openapi_fat_tck/fat/src/org/eclipse/microprofile/openapi/fat/tck/ConfigTckPackageTest.java]: Make sure you update the test name and class name in this.

File: dev/com.ibm.ws.microprofile.openapi_fat_tck/fat/src/org/eclipse/microprofile/openapi/fat/tck/ConfigTckPackageTest.java [hunk 10]
```diff
@@ -0,0 +1,88 @@
...
+        File tgt = new File(MvnUtils.home, "results/junit");
+
+        try {
+            Files.walkFileTree(src.toPath(), new MvnUtils.CopyFileVisitor(src.toPath(), tgt.toPath()));
+        } catch (java.nio.file.NoSuchFileException nsfe) {
+            Assert.assertNull(
+                    "The TCK tests' results directory does not exist which suggests the TCK tests did not run - check build logs."
+                            + src.getAbsolutePath(), nsfe);
+        }
+
```
Comment (navidsh) [inline dev/com.ibm.ws.microprofile.openapi_fat_tck/fat/src/org/eclipse/microprofile/openapi/fat/tck/ConfigTckPackageTest.java]: Please log what's the return code of the test run either 0 or not.

File: dev/com.ibm.ws.microprofile.openapi_fat_tck/bnd.bnd (hunk 1 above)
Comment (janamanoharan) [reply dev/com.ibm.ws.microprofile.openapi_fat_tck/bnd.bnd]: done

File: dev/com.ibm.ws.microprofile.openapi_fat_tck/fat/src/org/eclipse/microprofile/openapi/fat/tck/ConfigTckPackageTest.java (hunk 2 above)
Comment (janamanoharan) [reply dev/com.ibm.ws.microprofile.openapi_fat_tck/fat/src/org/eclipse/microprofile/openapi/fat/tck/ConfigTckPackageTest.java]: done

File: dev/com.ibm.ws.microprofile.openapi_fat_tck/fat/src/org/eclipse/microprofile/openapi/fat/tck/ConfigTckPackageTest.java (hunk 3 above)
Comment (janamanoharan) [reply dev/com.ibm.ws.microprofile.openapi_fat_tck/fat/src/org/eclipse/microprofile/openapi/fat/tck/ConfigTckPackageTest.java]: done

File: dev/com.ibm.ws.microprofile.openapi_fat_tck/fat/src/org/eclipse/microprofile/openapi/fat/tck/ConfigTckPackageTest.java (hunk 4 above)
Comment (janamanoharan) [reply dev/com.ibm.ws.microprofile.openapi_fat_tck/fat/src/org/eclipse/microprofile/openapi/fat/tck/ConfigTckPackageTest.java]: done

File: dev/com.ibm.ws.microprofile.openapi_fat_tck/fat/src/org/eclipse/microprofile/openapi/fat/tck/ConfigTckPackageTest.java (hunk 6 above)
Comment (janamanoharan) [reply dev/com.ibm.ws.microprofile.openapi_fat_tck/fat/src/org/eclipse/microprofile/openapi/fat/tck/ConfigTckPackageTest.java]: done

File: dev/com.ibm.ws.microprofile.openapi_fat_tck/fat/src/org/eclipse/microprofile/openapi/fat/tck/FATSuite.java (hunk 5 above)
Comment (janamanoharan) [reply dev/com.ibm.ws.microprofile.openapi_fat_tck/fat/src/org/eclipse/microprofile/openapi/fat/tck/FATSuite.java]: done

File: dev/com.ibm.ws.microprofile.openapi_fat_tck/fat/src/org/eclipse/microprofile/openapi/fat/tck/ConfigTckPackageTest.java (hunk 6 above)
Comment (janamanoharan) [reply dev/com.ibm.ws.microprofile.openapi_fat_tck/fat/src/org/eclipse/microprofile/openapi/fat/tck/ConfigTckPackageTest.java]: done

File: dev/com.ibm.ws.microprofile.openapi_fat_tck/fat/src/org/eclipse/microprofile/openapi/fat/tck/ConfigTckPackageTest.java (hunk 10 above)
Comment (janamanoharan) [reply dev/com.ibm.ws.microprofile.openapi_fat_tck/fat/src/org/eclipse/microprofile/openapi/fat/tck/ConfigTckPackageTest.java]: done

File: dev/com.ibm.ws.microprofile.openapi_fat_tck/fat/src/org/eclipse/microprofile/openapi/fat/tck/ConfigTckPackageTest.java (hunk 9 above)
Comment (janamanoharan) [reply dev/com.ibm.ws.microprofile.openapi_fat_tck/fat/src/org/eclipse/microprofile/openapi/fat/tck/ConfigTckPackageTest.java]: done

File: dev/com.ibm.ws.microprofile.openapi_fat_tck/bnd.bnd [hunk 11]
```diff
@@ -0,0 +1,32 @@
+#*******************************************************************************
+# Copyright (c) 2017 IBM Corporation and others.
```
Comment (navidsh) [inline dev/com.ibm.ws.microprofile.openapi_fat_tck/bnd.bnd]: Can you please update the year to 2018?

Comment (navidsh) [reply dev/com.ibm.ws.microprofile.openapi_fat_tck/bnd.bnd]: Please also update across the PR

File: dev/com.ibm.ws.microprofile.openapi_fat_tck/fat/src/org/eclipse/microprofile/openapi/fat/tck/OpenAPITckTest.java [hunk 12]
```diff
@@ -0,0 +1,94 @@
...
+        server.startServer();
+    }
+
+    @AfterClass
+    public static void tearDown() throws Exception {
+        server.stopServer("CWWKO1650E", "CWWKO1651W"); 
+    }
+
+    @Test
+    @AllowedFFDC // The tested deployment exceptions cause FFDC so we have to allow for this.
```
Comment (navidsh) [inline dev/com.ibm.ws.microprofile.openapi_fat_tck/fat/src/org/eclipse/microprofile/openapi/fat/tck/OpenAPITckTest.java]: Is this annotation needed to be here?

File: dev/com.ibm.ws.microprofile.openapi_fat_tck/publish/tckRunner/tck/LICENSE [hunk 13]
```diff
@@ -0,0 +1,201 @@
+                                 Apache License
```
Comment (navidsh) [inline dev/com.ibm.ws.microprofile.openapi_fat_tck/publish/tckRunner/tck/LICENSE]: We don't need the license file. Please delete this one

**Review by navidsh — APPROVED** [review]

**Pull request closed without merging**